packages/@aws-cdk/aws-servicecatalog/lib/tag-option.ts

```diff
@@ -0,0 +1,106 @@
+import * as cdk from '@aws-cdk/core';
+import { hashValues } from './private/util';
+import { InputValidator } from './private/validation';
+import { CfnTagOption } from './servicecatalog.generated';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct } from 'constructs';
+
+/**
+ * Properties for a service catalog TagOption.
+ */
+export interface TagOptionProps {
+  /**
+   * The TagOption key.
+   */
+  readonly key: string;
+
+  /**
+   * The TagOption value.
+   */
+  readonly value: string;
+}
+
+/**
+ * TagOption is a key-value pair managed in AWS Service Catalog.
+ * It is not an AWS tag, but serves as a template for creating an AWS tag based on the TagOption.
+ */
+export interface ITagOption extends cdk.IResource {
+  /**
+   * The ID of the TagOption.
+   * @attribute
+   */
+  readonly tagOptionId: string;
+}
+
+abstract class TagOptionBase extends cdk.Resource implements ITagOption {
+  public abstract readonly tagOptionId: string;
+}
+
+/**
+ * A Service Catalog TagOption.
+ */
+export class TagOption extends TagOptionBase {
+  /**
+   * Creates a TagOption construct that represents an external TagOption.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   */
+  public static fromTagOptionId(scope: Construct, id: string, tagOptionId: string): ITagOption {
+    class Import extends TagOptionBase {
+      public readonly tagOptionId = tagOptionId;
+    }
+    return new Import(scope, id);
+  }
+
+  public readonly tagOptionId: string;
+
+  constructor(scope: Construct, id: string, props: TagOptionProps) {
+    super(scope, id);
+    InputValidator.validateLength(this.node.path, 'TagOption key', 1, 128, props.key);
+    InputValidator.validateLength(this.node.path, 'TagOption value', 1, 256, props.value);
+
+    const tagOptionKey = hashValues(props.key, props.value);
+    const cfnTagOption = new CfnTagOption(this, `TagOption${tagOptionKey}`, {
+      key: props.key,
+      value: props.value,
+    });
+    this.tagOptionId = cfnTagOption.ref;
+  }
+}
+
+/**
+ * Defines a set of TagOptions, which are a list of key-value pairs managed in AWS Service Catalog.
+ * It is not an AWS tag, but serves as a template for creating an AWS tag based on the TagOption.
+ */
+export class TagOptions {
+  /**
+   * Map of underlying TagOption resources.
+   */
+  public readonly tagOptionsMap: { [key:string] : {[value:string] :TagOption}};
+
+  constructor(scope: Construct, tagOptions: { [key: string]: string[] }) {
+    this.tagOptionsMap = this.createUnderlyingTagOptions(scope, tagOptions);
+  }
+
+  private createUnderlyingTagOptions(scope: Construct, tagOptions: { [key: string]: string[] }): {[key:string] : {[value:string] :TagOption}} {
+    var tagOptionMap: {[key:string] : {[value:string] : TagOption}} = {};
```

Comment:
I did this originally because it was floated that we should be able to extract an individual tagOption out of the aggregate resource, in which case we would need both the key and value to reference.

Reply:
But I now think from UX perspective, if users are trying to extract a certain key-value or key from tagOptions, they should just instead instantiate that as a separate class, then we can get rid of this and make it a simple map of the hashkey to the underlying tagOption.